catalog/task/push-snapshot/0.4/push-snapshot.yaml

```diff
@@ -0,0 +1,68 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: push-snapshot
+  labels:
+    app.kubernetes.io/version: "0.3"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: release
+spec:
+  description: >-
+    Tekton task to push snapshot images to an image registry using `skopeo copy`
+  params:
+    - name: mappedSnapshot
+      description: JSON string representing the Snapshot
+      type: string
+    - name: tag
+      description: tag name to create on external registry
+      type: string
+      default: "latest"
+    - name: retries
+      description: Retry copy N times.
+      type: string
+      default: "0"
+  steps:
+    - name: skopeo-copy
+      image:
+        quay.io/hacbs-release/release-base-image@sha256:9e7fd1a3ccf0d2c8077f565c78e50862a7cc4792d548b5c01c8b09077e6d23a7
+      script: |
+        #!/usr/bin/env bash
+        set -eux
+        application=$(jq -r '.application' <<<'$(params.mappedSnapshot)')
+        printf 'Beginning "%s" for "%s"\n\n' "$(context.task.name)" "$application"
+        while read line;
+        do
+          # Create array with component values
+          typeset -A data
+          while IFS== read -r key value; do
+            data["$key"]="$value"
+          done < <(jq -r '. | to_entries | .[] | .key + "=" + .value' <<<"$line")
+          data[tag]="${data[tag]:=$(params.tag)}"
```

Comment:
Nit: `:-` would probably be more appropriate than `:=` here. `:=` already assigns the value to the variable (if undefined or empty) - see here: https://tldp.org/LDP/Bash-Beginners-Guide/html/sect_10_03.html

But maybe better not to touch it if it works :)

I want to ask about something else. Can you explain what are the circumstances when the tag is set in the snapshot data vs. when it's not?

The tag parameter description says: "tag name to create on external registry", but this will only be used if the snapshot json does not contain a tag string. So I'm trying to understand when that happens.

Reply:
The second part of my comment above is sorted - it was explained elsewhere.

The nitpick is still valid and it's up to you if you want to change it or not. Feel free to just resolve this.